### PR TITLE
Implement JSON Import/Export Improvements (#242)

### LINKED ISSUE #242: Adjustments to JSON import and export
When exporting or importing activities the JSON field exposes a field named `isActive`. I'm not sure where this is used, as it's not visible in the form when creating or editing activities. Perhaps we don't need it? Also, the import should allow incomplete JSON objects where only the bare necessities are present in an activity. What I mean with that is if a user uploads a JSON file where the objects in the array only contain name, the rest should be automatically populated.

Example of the kind of JSON I want to be valid:

```json
[
  {
    "id": "1",
    "name": "Homework",
    "description": "Academic work and study time",
    "colorIndex": 0,
    "createdAt": "2025-07-17T10:19:38.837Z",
    "isActive": true
  },
  {
    "name": "Reading",
    "description": "Reading books, articles, or educational materials",
    "colorIndex": 1
  },
  {
    "name": "Play Time"
  },
  {
    "id": "4",
    "name": "Chores"
  },
  {
    "name": "Exercise",
    "colorIndex": 4
  },
  {
    "id": "6",
    "name": "Creative Arts",
    "description": "Drawing, painting, and other creative pursuits",
    "colorIndex": 5
  }
]
```

Importing this JSON structure should automatically assign the missing or empty fields, and keep in mind they might either be missing, undefined or null.

### PR DESCRIPTION
**Pull request by ken-guru**
## Overview
This PR implements comprehensive enhancements to the activity JSON import/export functionality as requested in issue #242.

### Enhanced Export Functionality
- ✅ **Clean JSON exports**: Removes `isActive` field by default for cleaner user-facing JSON
- ✅ **Configurable exports**: Optional `includeIsActive` and `activeOnly` parameters
- ✅ **Backwards compatibility**: Existing workflows continue to work unchanged

### Enhanced Import Functionality  
- ✅ **Flexible imports**: Accepts minimal JSON objects with only `name` field required
- ✅ **Intelligent auto-population**: Missing fields filled with sensible defaults:
  - `id`: Generated UUID or timestamp-based ID
  - `description`: Handled as optional field
  - `colorIndex`: Assigned next available color to avoid conflicts
  - `createdAt`: Current timestamp if missing
  - `isActive`: Defaults to true for new activities
- ✅ **Enhanced validation**: Clear error messages for invalid import data
- ✅ **Backwards compatibility**: Existing export files import without issues

### New Files
- `src/utils/activity-import-export.ts` - Core import/export utilities with TypeScript interfaces
- `src/utils/activity-import-export.test.ts` - Comprehensive test coverage (18 tests)

### Modified Files
- `src/components/feature/ActivityCrud.tsx` - Updated to use new utilities with enhanced error handling
- `docs/PLANNED_CHANGES.md` - Cleared after completion
- `docs/IMPLEMENTED_CHANGES.md` - Added implementation record

## Test Coverage
- ✅ **18 comprehensive Jest tests** covering all import/export scenarios
- ✅ **Existing ActivityCrud tests (15)** continue to pass
- ✅ **Edge cases covered**: Invalid data, missing fields, empty arrays, malformed JSON
- ✅ **Integration scenarios**: Complete export-import cycles

## Quality Assurance
- ✅ All tests passing (33/33)
- ✅ Linting successful  
- ✅ TypeScript type checking successful
- ✅ TDD methodology followed (tests written first)

## User Experience Improvements
- **Cleaner exports**: JSON files no longer contain internal `isActive` field
- **Easier imports**: Users can import minimal JSON with just activity names
- **Better error messages**: Clear, actionable feedback for import validation failures
- **Maintained workflows**: No breaking changes to existing functionality

### Clean Export (New)
```json
[
  {
    "id": "1",
    "name": "Morning Routine", 
    "colorIndex": 0,
    "createdAt": "2025-01-15T10:00:00.000Z"
  }
]
```

### Minimal Import (New)
```json
[
  { "name": "New Activity" }
]
```
Auto-populates: id, colorIndex, createdAt, isActive

Resolves #242

### CODE REVIEW

File: src/utils/activity-import-export.ts [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+      id = crypto.randomUUID ? crypto.randomUUID() : `imported-${Date.now()}-${index}`;
+    }
+    
+    // Use provided colorIndex or assign next available
+    let colorIndex = partialActivity.colorIndex;
+    if (typeof colorIndex !== 'number' || colorIndex < 0) {
+      colorIndex = getNextColorIndex();
+    } else {
+      // Mark provided colorIndex as used
+      usedColorIndices.add(colorIndex);
```
Comment (ken-guru) [inline src/utils/activity-import-export.ts:154]: ✅ **TDD Comment Addressed**

You're absolutely right about following TDD approach. I've now added comprehensive test coverage for the `crypto.randomUUID` fallback scenario:

**Added Test Case:** `should use fallback ID generation when crypto.randomUUID is unavailable`

**Test Coverage:**
- Mocks `crypto.randomUUID` to be `undefined` 
- Verifies fallback logic uses timestamp-based ID format: `imported-{timestamp}-{index}`
- Tests with fixed timestamp for deterministic assertions
- Properly restores mocks after test execution

**Current Test Results:** 19/19 tests passing including the new fallback test

The fallback logic is now properly tested with both available and unavailable `crypto.randomUUID` scenarios, ensuring robust behavior across different environments.

File: src/utils/activity-import-export.ts [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    
+    // Use provided isActive or default to true
+    const isActive = typeof partialActivity.isActive === 'boolean' 
+      ? partialActivity.isActive 
+      : true;
+    
+    // Handle description (can be undefined, null, or string)
+    const description = partialActivity.description === null 
+      ? undefined 
+      : partialActivity.description;
```
[the remark this answers is not in the document]
Comment (ken-guru) [reply src/utils/activity-import-export.ts]: ✅ **Code Maintainability Comment Addressed**

Thank you for the suggestion! I've implemented the `normalizeDescription()` helper function as recommended:

**Implementation:**
```typescript
/**
 * Normalize description field handling null/undefined values
 * @param description - Description value from import data
 * @returns Normalized description value
 */
function normalizeDescription(description?: string | null): string | undefined {
  return description === null ? undefined : description;
}
```

**Usage in importActivities:**
```typescript
// Handle description using helper function
const description = normalizeDescription(partialActivity.description);
```

This extraction improves:
- **Readability**: Complex conditional logic is now encapsulated in a named function
- **Maintainability**: Single responsibility principle - description normalization logic is isolated
- **Testability**: The helper function can be independently tested if needed

The logic now clearly communicates its intent: "normalize description values by converting null to undefined while preserving other values."
